client/src/Components/ResultsTable.js

```diff
@@ -0,0 +1,114 @@
+import React from "react";
+import { useTable } from "react-table";
+const [results,setAllResults]=useState();
+const Styles = styled.div`
+  padding: 1rem;
+
+  table {
+    border-spacing: 0;
+    border: 1px solid black;
+
+    tr {
+      :last-child {
+        td {
+          border-bottom: 0;
+        }
+      }
+    }
+
+    th,
+    td {
+      margin: 0;
+      padding: 0.5rem;
+      border-bottom: 1px solid black;
+      border-right: 1px solid black;
+
+      :last-child {
+        border-right: 0;
+      }
+    }
+  }
+`;
+export default function ResultsTable() {
+	const [studentNames, setStudentNames]=useState([]);
+	const [selQuizQuestions, setSelQuizQuestions]=useState([]);
+	const [answeredQuestons, setAnsweredQuestons]=useState([]);
+	const [allResults, setAllResults]=useState(null);
+	const [quizSelected, setQuizSelected]=useState(null);
+	const fetchResults =()=>{
+		fetch("http://localhost:3100/api/results/")
+			.then((response)=>response.json())
+			.then((data)=>{
+				setAllResults(data);
+				console.log({ data });
+			});
+	};
+	useEffect(() => {
+		fetchResults();
+	},[]);
+	useEffect(() => {
+
+
+		if(quizSelected!==null){
+			const makeQuestions=()=>{
+				let selectedQuizQuestions= [];
+				selectedQuizQuestions = quizSelected.questions_id.map((selId)=>{
+
+					let found = (props.questions.find((question)=>question._id==selId));
+					console.log({ found });
+					selectedQuizQuestions.push(found);
+					setSelQuizQuestions(selectedQuizQuestions);
+				}
+				);
+
+			};
+			makeQuestions();
+		}
+
+	},[quizSelected]);
+	function Table({ columns, data }) {
+		// Use the state and functions returned from useTable to build your UI
+		const {
+			getTableProps,
+			getTableBodyProps,
+			headerGroups,
+			rows,
+			prepareRow,
+		} = useTable({
+			columns,
+			data,
+		});
+
+		// Render the UI for your table
+		return (
+			<table {...allResults}>
+				<thead>
+					{headerGroups.map((headerGroup) => (
+						<tr {...headerGroup.getHeaderGroupProps()}>
+							{headerGroup.headers.map((column) => (
+								<th {...column.getHeaderProps()}>{column.render("Header")}</th>
+							))}
+						</tr>
+					))}
+				</thead>
+				<tbody {...getTableBodyProps()}>
+					{rows.map((row, i) => {
+						prepareRow(row);
+						return (
+							<tr {...row.getRowProps()}>
+								{row.cells.map((cell) => {
+									return <td {...cell.getCellProps()}>{cell.render("Cell")}</td>;
+								})}
+							</tr>
+						);
```

Comment:
I haven't seen this type of coding
I can not review it

Reply:
This one is not going to be merged at least in upcoming days... I mention it in proposal in the description of pull request